sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -3138,13 +3141,54 @@ class CSVv1Suite extends CSVSuite {
     super
       .sparkConf
       .set(SQLConf.USE_V1_SOURCE_LIST, "csv")
+
+  private val carsFile = "test-data/cars.csv"
+
+  test("test for FAILFAST parsing mode on CSV v1") {
+    Seq(false, true).foreach { multiLine =>
+      val exception = intercept[SparkException] {
+        spark.read
+          .format("csv")
+          .option("multiLine", multiLine)
+          .options(Map("header" -> "true", "mode" -> "failfast"))
+          .load(testFile(carsFile)).collect()
+      }
+
+      checkError(
+        exception = exception.getCause.asInstanceOf[SparkException],
+        errorClass = "_LEGACY_ERROR_TEMP_2177",
```

Comment:
Could you explain why did you add the test for the error class in the PR.

Reply:
I covered it as well in this PR because it's in the same error chain. it is found by: https://github.com/apache/spark/pull/39258#discussion_r1058263385